frontends/web/src/routes/device/bitbox02/mnemonicpassphrase.tsx

```diff
@@ -16,78 +16,89 @@
  */
 
 import { Component, h, RenderableProps } from 'preact';
+import { getDeviceInfo, setMnemonicPassphraseEnabled, TDeviceInfo } from '../../../api/bitbox02';
 import { translate, TranslateProps } from '../../../decorators/translate';
-import { apiPost } from '../../../utils/request';
 import { SimpleMarkup } from '../../../utils/simplemarkup';
 import { alertUser } from '../../../components/alert/Alert';
 import { SettingsButton } from '../../../components/settingsButton/settingsButton';
 import { WaitDialog } from '../../../components/wait-dialog/wait-dialog';
 
 interface MnemonicPassphraseButtonProps {
-    apiPrefix: string;
-    getInfo: () => void;
-    mnemonicPassphraseEnabled: boolean;
+    deviceID: string;
+    deviceInfo: TDeviceInfo;
 }
 
 interface State {
     inProgress: boolean;
+    deviceInfo: TDeviceInfo;
 }
 
 type Props = MnemonicPassphraseButtonProps & TranslateProps;
 
 class MnemonicPassphraseButton  extends Component<Props, State> {
+    public readonly state: State = {
+        inProgress: false,
+        deviceInfo: this.props.deviceInfo,
+    }
+
+    private getDeviceInfo = () => {
+        getDeviceInfo(this.props.deviceID).then((response) => {
+            if (response.success) {
+                this.setState({ deviceInfo: response.deviceInfo });
+            }
+        });
+    }
+
     private toggle = () => {
-        const enable = !this.props.mnemonicPassphraseEnabled;
+        const enable = !this.state.deviceInfo.mnemonicPassphraseEnabled;
         this.setState({ inProgress: true });
-        apiPost(this.props.apiPrefix + '/set-mnemonic-passphrase-enabled',
-                enable).then(({ success }) => {
-                    this.setState({ inProgress: false });
-                    if (success) {
-                        if (enable) {
-                            alertUser(this.props.t('bitbox02Settings.mnemonicPassphrase.successEnable'));
-                        } else {
-                            alertUser(this.props.t('bitbox02Settings.mnemonicPassphrase.successDisable'));
-                        }
-                        this.props.getInfo();
+        setMnemonicPassphraseEnabled(this.props.deviceID, enable)
+            .then((response) => {
```

Comment:
i think the parens around (response) are not needed

before the change it was `({ success })`, which is also pretty nice I think

Reply:
Yes not needed, but also `;` is not needed and we still write it, see ASI https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#automatic_semicolon_insertion

Having it always is IMO easier to read and easier to change the code again in the future, i.e. adding arg destruction or adding a second arg.

I change it for now, but in the future I would like to just stick to what react-scripts suggest